### PR TITLE
Survey: Date picker doesn't work on IE

### PR DESCRIPTION
**Pull request by glorymanutd**
#1629

### CODE REVIEW

File: waltz-ng/client/survey/components/survey-run-create-general.js [hunk 1]
```diff
@@ -31,9 +31,26 @@ const initialState = {};
 const template = require('./survey-run-create-general.html');
 
 
+function initialiseDatePicker(vm) {
+    vm.dateOptions = {
+        formatYear: 'yyyy',
+        maxDate: new Date(2020, 5, 22),
```
Comment (davidwatkins73) [inline waltz-ng/client/survey/components/survey-run-create-general.js]: this needs to be dynamic

File: waltz-ng/client/survey/components/survey-run-create-general.html [hunk 2]
```diff
@@ -74,10 +74,24 @@
...
+                           datepicker-options="$ctrl.dateOptions"
+                           close-text="Close" />
+                    <span class="input-group-btn">
+						<button type="button"
+                                class="btn btn-default"
+                                ng-click="$ctrl.dueDatePickerOpen()">
+							<waltz-icon name="calendar"  />
+						</button>
+					</span>
+                </p>
```
Comment (davidwatkins73) [inline waltz-ng/client/survey/components/survey-run-create-general.html]: Formatting a bit funky

**Comment by davidwatkins73** [comment]
Does uib-datepicker offer any advantage over the built in date pickers in modern browsers.  Should we hide this behind a widget and make it IE only ?

File: waltz-ng/client/survey/components/survey-run-create-general.js (hunk 1 above)
Comment (glorymanutd) [inline waltz-ng/client/survey/components/survey-run-create-general.js]: Removed the maxDate

**Comment by glorymanutd** [comment]
I've moved the date-picker into a separate component.

I think some advantages of uib-datepicker are:

- consistent look across Chrome and IE
- always showing the 'calendar' button so users know that the can click and select the date where with input type=date the dropdown icon only appear when user's on the input
- flexible date formats where the html4 date control always takes browser locale format

**Pull request merged**